File: CodingInterviewsPythonSolutions/Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sz = len(nums)
        result = float('-inf')
        maxSoFar = 0
        
        for i in range(sz):
            maxSoFar = max(nums[i], nums[i] + maxSoFar)
            result = max(result, maxSoFar)
        
        return result

File: CodingInterviewsPackSolutionsCPP/Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sz = nums.size();
        int result = INT_MIN;
        int maxSoFar = 0;
        
        for(int i = 0; i < sz; i++){
            maxSoFar = max(nums[i],nums[i]+maxSoFar);
            result = max(result,maxSoFar);
        }
        return result;
    }
};
